tests/emitter/emitter.js

```diff
@@ -0,0 +1,356 @@
+
+		sinon.assert.calledThrice( spy );
+	} );
+} );
+
+describe( 'on', function() {
+	it( 'should stop()', function() {
+		var spy1 = sinon.spy();
+		var spy2 = sinon.spy();
+		var spy3 = sinon.spy( function( event ) {
+			event.stop();
+		} );
+
+		emitter.on( 'test', spy1 );
+		emitter.on( 'test', spy2 );
+		emitter.on( 'test', spy3 );
+		emitter.on( 'test', sinon.stub().throws() );
+		emitter.on( 'test', sinon.stub().throws() );
+
+		emitter.fire( 'test' );
+
+		sinon.assert.called( spy1 );
+		sinon.assert.called( spy2 );
+		sinon.assert.called( spy3 );
+	} );
+
+	it( 'should take a callback off()', function() {
+		var spy1 = sinon.spy();
+		var spy2 = sinon.spy( function( event ) {
+			event.off();
+		} );
+		var spy3 = sinon.spy();
+
+		emitter.on( 'test', spy1 );
+		emitter.on( 'test', spy2 );
+		emitter.on( 'test', spy3 );
+
+		emitter.fire( 'test' );
+		emitter.fire( 'test' );
+
+		sinon.assert.calledTwice( spy1 );
+		sinon.assert.calledOnce( spy2 );
+		sinon.assert.calledTwice( spy3 );
+	} );
+} );
+
+describe( 'once', function() {
+	it( 'should be called just once', function() {
+		var spy1 = sinon.spy();
+		var spy2 = sinon.spy();
+		var spy3 = sinon.spy();
+
+		emitter.on( 'test', spy1 );
+		emitter.once( 'test', spy2 );
+		emitter.on( 'test', spy3 );
+
+		emitter.fire( 'test' );
+		emitter.fire( 'test' );
+
+		sinon.assert.calledTwice( spy1 );
+		sinon.assert.calledOnce( spy2 );
+		sinon.assert.calledTwice( spy3 );
+	} );
+
+	it( 'should have proper scope', function() {
+		var ctx = {};
+
+		var spy1 = sinon.spy();
+		var spy2 = sinon.spy();
+
+		emitter.once( 'test', spy1, ctx );
+		emitter.once( 'test', spy2 );
+
+		emitter.fire( 'test' );
+
+		sinon.assert.calledOn( spy1, ctx );
+		sinon.assert.calledOn( spy2, emitter );
+	} );
+
+	it( 'should have proper arguments', function() {
+		var EventInfo = modules.eventinfo;
+
+		var spy = sinon.spy();
+
+		emitter.once( 'test', spy );
+
+		emitter.fire( 'test', 1, 2, 3 );
+
+		sinon.assert.calledWithExactly( spy, sinon.match.instanceOf( EventInfo ), 1, 2, 3 );
+	} );
+} );
+
+describe( 'off', function() {
+	it( 'should get callbacks off', function() {
+		var spy1 = sinon.spy();
+		var spy2 = sinon.spy();
+		var spy3 = sinon.spy();
+
+		emitter.on( 'test', spy1 );
+		emitter.on( 'test', spy2 );
+		emitter.on( 'test', spy3 );
+
+		emitter.fire( 'test' );
+
+		emitter.off( 'test', spy2 );
+
+		emitter.fire( 'test' );
+		emitter.fire( 'test' );
+
+		sinon.assert.calledThrice( spy1 );
+		sinon.assert.calledOnce( spy2 );
+		sinon.assert.calledThrice( spy3 );
+	} );
+
+	it( 'should no fail with unknown events', function() {
+		emitter.off( 'test', function() {} );
+	} );
+} );
+
+describe( 'listenTo', function() {
+	beforeEach( refreshListener );
+
+	it( 'should properly register callbacks', function() {
+		var spy = sinon.spy();
+
+		listener.listenTo( emitter, 'test', spy );
+
+		emitter.fire( 'test' );
+
+		sinon.assert.called( spy );
+	} );
+} );
+
+describe( 'stopListening', function() {
```

Comment:
Is this what you meant for this?

```js
it( 'should not stop other emitters when a non-listened emitter is provided', function() {
	var spy = sinon.spy();

	var emitter1 = getEmitterInstance();
	var emitter2 = getEmitterInstance();

	listener.listenTo( emitter1, 'test', spy );

	listener.stopListening( emitter2 );

	emitter1.fire( 'test' );

	sinon.assert.called( spy );
} );
```

It goes green for me, but from your comments I had the impression that you expected it to be red. Maybe you want a different test. Lemme know.